client/examples/Main.purs

```diff
@@ -0,0 +1,66 @@
+module Main where
+
+import Prelude
+
+import Data.Foldable (fold)
+import Effect (Effect)
+import TryPureScript (h1, h2, p, text, list, indent, link, render, code)
+
+main :: Effect Unit
+main =
+    render $ fold
+      [ h1 (text "Try PureScript!")
+      , p (text "Try out the examples below, or create your own!")
+      , h2 (text "Examples")
+      , list (map fromExample examples)
+      , h2 (text "Share Your Code")
+      , p (text "A PureScript file can be loaded from GitHub from a gist or a repository. To share code using a gist, simply include the gist ID in the URL as follows:")
+      , indent (p (code (text "  try.purescript.org?gist=gist-id")))
+      , p (fold
+          [ text "The Gist should contain PureScript modulenamed "
+          , code (text "Main")
+          , text "in a file named "
+          , code (text "Main.purs")
+          , text " containing your PureScript code."
+          ])
+      , p (text "To share code from a repository, include the path to the source file the URL as follows:")
+      , indent (p (code (text "  try.purescript.org?github=/owner/repo/Source.purs")))
+      , p (fold
+          [ text "The file should be a PureScript module named "
+          , code (text "Main")
+          , text " containing your PureScript code."
+          ])
+      ]
+  where
+    fromExample { title, source } =
+      link ("https://github.com/purescript/trypurescript/load-from-github/client/examples/" <> source) (text title)
```

Comment:
I think this ought to use a tag from some kind of configuration rather than hardcoding the branch name, so that we can change these without breaking the currently released version. In practice we will want this to be the tag from the current version.

Reply:
Yea — I could thread this in as an argument to main, so that it can be accessed in various places taken from outside the PureScript code.

Reply:
Sounds good :+1:

Reply:
I could do that with the entire dev / prod configurations in that case

Reply:
If that makes the overall setup simpler, sure, I don't see why not.

Reply:
On second thought, doing this would require some kind of text replacement processing pass to work. After all, this example is code loaded literally into the Try PureScript editor.

Otherwise, we'd have to do something like read the current tag from data in the index.html file or something like that, but this would be user-visible code and I'm not a huge fan of that.

Another option is to manually hard-code the tag as part of the release process and document where the tag would need to be updated -- we release so infrequently that this feels legitimately like an option.

Reply:
True - I think manually hardcoding the tag and documenting that it needs updating will work.

Reply:
Perhaps we should split the repository README so that the main README is for users and there's a separate development guide. It could include instructions for developing, like we already have, but could also have a release guide which includes:

- Updating the hardcoded tag
- Checking externs sizes
- Running the deployment

What do you think?

Reply:
For now I've just added a note about the hardcoded tags to the README, but I think we probably do want a dedicated, separate development + release guide.